cli/src/lib.rs

```diff
@@ -21,3 +21,27 @@ extern crate thiserror;
 
 pub mod commands;
 pub mod helpers;
+
+use anyhow::Result;
+use std::path::PathBuf;
+
+pub fn check_parent_permissions(path: &PathBuf) -> Result<()> {
```

Comment:
```suggestion
pub fn check_parent_permissions<T: AsRef<Path>>(path: T) -> Result<()> {
```
this should allow both `&str` (or `&String`) and `&PathBuf` to work, making `check_parent_permissions_for_string` redundant

Reply:
Great suggestion. Fixed :sun_with_face: